### ISSUE TITLE
Default number of blocks for the address cache command.

### ISSUE THREAD

**Issue by buhtignew** [slimcoin-project/pacli#90]
I've been testing `address cache mx5MdsenFDZufFuwT9ND7BKQBzS4Hy9YUP -s 156492` command and has discovered that regardless whether I put the `-b` flag or not the considers the default value of 50000 blocks.

In fact in the output of the above command was reported that the block actually processed were those from 156500 to 206400.

Is it the intended behavior?

**Comment by d5000** [comment]
Here it works perfectly, even with your address. Syntax is `address cache ADDRESS -s STARTBLOCK -b NUMBER_OF_BLOCKS`.

**Comment by buhtignew** [comment]
It works fine on my side as well.

The only question I have is whether the `-b=50000` is implicitly attached to the command even if there is no `-b` flag added by the user. 

My expectation were:
1) there would be no limit in case there is no `-b` flag,
2) there would be 50000 limit in case a `-b` flag included,
3) there would be a limit set up by the user in case of `-b=[the_user's_limit]` flag included.

Above all I think there should be a way to set up an unlimited caching, if not the user would need to launch the command many times in the production blockchain to get all the data which may be not comfortable.

**Comment by d5000** [comment]
Yes, 50000 is the default, this means if no number is given, 50000 will be scanned. I chose this behavior because the whole blockchain on mainnet would take days to scan.

Nevertheless I added the option `--full / -f`, which will scan the whole blockchain (from the startblock on), because it is a low hanging fruit and there may be people wanting to scan several addresses overnight, and it can be interrupted / continued at any time. It was also added to `deck/token cache`. Commit: e7537ce

**Comment by buhtignew** [comment]
Everything fine with this issue. I'm closing this.
